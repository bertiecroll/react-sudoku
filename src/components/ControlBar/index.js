export { ControlBar } from './ControlBar'
export { default } from './ControlBar'
