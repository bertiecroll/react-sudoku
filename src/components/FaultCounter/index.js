export { FaultCounter } from './FaultCounter'
export { default } from './FaultCounter'
