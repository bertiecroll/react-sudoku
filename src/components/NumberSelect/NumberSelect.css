.NumberSelect {
  margin: 1em;
  padding: .25em;
  display: flex;
  flex-direction: row;
}
