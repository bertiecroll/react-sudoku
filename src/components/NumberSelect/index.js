export { NumberSelect } from './NumberSelect'
export { default } from './NumberSelect'
