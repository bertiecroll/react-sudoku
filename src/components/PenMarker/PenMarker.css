.PenMarker .NumberSelect .NumberButton {
  background-color: #67DAF9;
  color: white;
}
