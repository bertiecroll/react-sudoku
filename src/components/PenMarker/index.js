export { PenMarker } from './PenMarker'
export { default } from './PenMarker'
