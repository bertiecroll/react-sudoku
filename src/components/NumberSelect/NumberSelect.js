import React from 'react'
import PropTypes from 'prop-types'
import './NumberSelect.css'

export const NumberSelect = (props) => {
  return (
    <div className="NumberSelect">
      {props.availableNumbers.map(number => <div key={number} onClick={() => props.handleClick(number)}>{number}</div>)}
    </div>
  )
}

NumberSelect.propTypes = {
  availableNumbers: PropTypes.array.isRequired
}
