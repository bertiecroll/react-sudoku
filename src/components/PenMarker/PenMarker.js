import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import './PenMarker.css'
import NumberSelect from 'components/NumberSelect'
import { incrementFaultCounter, updateFaultCellId } from 'redux/actions/fault'
import { markCellCompleted } from 'redux/actions/grid'

export function PenMarker ({ markCellCompleted, incrementFaultCounter, updateFaultCellId }) {

  const onClickAction = (selectedCell, selectedNumber) => {
    const { id, value } = selectedCell
    value === selectedNumber ? markCellCompleted(id) : showTempFaultAlert(id)
  }

  const showTempFaultAlert = id => {
    incrementFaultCounter()
    updateFaultCellId(id)
    setTimeout(() => updateFaultCellId(null), 1000)
  }

  return (
    <div className="PenMarker">
      <NumberSelect onClickAction={onClickAction} />
    </div>
  )
}

const mapDispatchToProps = { markCellCompleted, incrementFaultCounter, updateFaultCellId }
export default connect(null, mapDispatchToProps)(PenMarker)

PenMarker.propTypes = {
  markCellCompleted: PropTypes.func.isRequired,
  incrementFaultCounter: PropTypes.func.isRequired,
  updateFaultCellId: PropTypes.func.isRequired,
}
