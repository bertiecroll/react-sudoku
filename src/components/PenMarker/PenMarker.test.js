import React from 'react'
import { shallow } from 'enzyme'
import { PenMarker } from 'components/PenMarker'
import NumberSelect from 'components/NumberSelect'

describe('PenMarker', () => {
  let wrapper
  const mockMarkCellCompleted = jest.fn()
  const mockIncrementFaultCounter = jest.fn()
  const mockUpdateFaultCellId = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <PenMarker
        markCellCompleted={mockMarkCellCompleted}
        incrementFaultCounter={mockIncrementFaultCounter}
        updateFaultCellId={mockUpdateFaultCellId}
      />
    )
  })

  afterEach(() => {
    mockMarkCellCompleted.mockClear()
    mockIncrementFaultCounter.mockClear()
    mockUpdateFaultCellId.mockClear()
  })

  it('renders NumberSelect component passing onClickAction', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('#onClickAction', () => {
    const selectedCell = { id: '0,0', value: 1 }

    it('calls #markCellCompleted when selectedCell value is equal to selectedNuber', () => {
      const selectedNumber = 1

      wrapper.find(NumberSelect).props().onClickAction(selectedCell, selectedNumber)

      expect(mockMarkCellCompleted).toHaveBeenCalledWith(selectedCell.id)
    })

    it('calls #incrementFaultCounter and #updateFaultCellId when selectedCell value is not equal to selectedNuber', () => {
      const selectedNumber = 2

      wrapper.find(NumberSelect).props().onClickAction(selectedCell, selectedNumber)

      expect(mockIncrementFaultCounter).toHaveBeenCalled()
      expect(mockUpdateFaultCellId).toHaveBeenCalledWith(selectedCell.id)
    })
  })
})
