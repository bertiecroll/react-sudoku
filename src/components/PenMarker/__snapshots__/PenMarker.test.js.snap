// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PenMarker renders NumberSelect component passing onClickAction 1`] = `
<div
  className="PenMarker"
>
  <Connect(NumberSelect)
    onClickAction={[Function]}
  />
</div>
`;
